jupyterhub/spawner.py

```diff
@@ -495,7 +495,7 @@ def get_args(self):
         """
         args = [
             '--user="%s"' % self.user.name,
-            '--base-url="%s"' % self.user.server.base_url,
```

Comment:
server.base_url is already defined, as seen in this line. Since each Spawner now needs to have a different `server`, it should be `self.server` instead of `self.user.server` here.